lib/import-and-filter-project.nix

```diff
@@ -37,9 +37,21 @@ in project // {
                       then
                         pkgs.lib.lists.elemAt sourceRepos (
                           pkgs.lib.strings.toInt (pkgs.lib.strings.removePrefix srcRepoPrefix subDir))
-                      else if haskellLib.canCleanSource srcRoot
-                        then haskellLib.cleanSourceWith { src = srcRoot; inherit subDir; }
-                        else srcRoot + (if subDir == "" then "" else "/" + subDir);
+                      else (if haskellLib.canCleanSource srcRoot
+                        then haskellLib.cleanSourceWith {
+                          src = srcRoot;
+                          inherit subDir;
+                          includeSiblings = true; # Filtering sibling dirs of the package dir is done in the
+                                                  # component builder so that relative paths can be used to
+                                                  # reference project directories not in the package subDir.
+                        }
+                        else {
+                          outPath = srcRoot + (if subDir == "" then "" else "/" + subDir);
+                        }) // {
+                          # Tag the `src` attribute of the packages that are part of the project
+                          # (but not source repository packages)
+                          isProjectPackageSrc = true;
```

Comment:
This seems useful - but I think maybe we should try and get this as a package attribute instead of a source attribute?

Reply:
Changed to a module option.